### PR TITLE
Two tweaks to shared tests

### PR DESCRIPTION
**Pull request by mlafeldt**
- Don't panic when type assertions fail
- Test that store is empty before adding anything

### CODE REVIEW

File: shared_test/feature_store.go [hunk 1]
```diff
@@ -33,8 +33,10 @@ func RunFeatureStoreTests(t *testing.T, makeStore func() ld.FeatureStore) {
 		assert.NotNil(t, result)
 		assert.NoError(t, err)
 
-		r := result.(*ld.FeatureFlag)
-		assert.Equal(t, feature1.Key, r.Key)
+		r, ok := result.(*ld.FeatureFlag)
```
Comment (ashanbrown) [inline shared_test/feature_store.go]: I'm all for test suites that don't panic.  Testify has a method that does this if you think that might be cleaner. 
 It's at youhttps://godoc.org/github.com/stretchr/testify/assert#IsType

**Comment by mlafeldt** [comment]
The problem is that testify doesn't abort if a test fails, no matter what
assertion is used. Will check out IsType once I'm back at my laptop. 👍

**Comment by mlafeldt** [comment]
@ashanbrown I've pushed an update - using IsType as suggested. PTAL.

**Comment by ashanbrown** [comment]
Thanks @mlafeldt.  Honestly, I'm not sure `IsType` was a good suggestion -- it looks like we have to do the type-cast no matter way -- but either way, this is a helpful addition.  I'll merge it.

**Comment by mlafeldt** [comment]
@ashanbrown Yeah, I actually felt the same when making the change. 😄

**Pull request merged**